CMakeLists.txt

```diff
@@ -0,0 +1,255 @@
+        DESTINATION ${CONFIG_INSTALL_DIR}
+)
+
+# build directory package config
+export(EXPORT ${TARGETS_EXPORT_NAME}
+       FILE ${PROJECT_CONFIG_FILE}
+)
+
+# installed package config
+install(EXPORT ${TARGETS_EXPORT_NAME}
+        DESTINATION ${CONFIG_INSTALL_DIR}
+        FILE ${PROJECT_CONFIG_FILE}
+)
+
+# add to CMake registry
+export(PACKAGE ${PROJECT_NAME})
+
+#---------------------------------------------------------------------------------------
+# Create release packages
+#---------------------------------------------------------------------------------------
+# package release version
+set(PACKAGE_RELEASE_VERSION 1)
+
+# determine distribution and architecture
+find_program(LSB_RELEASE lsb_release)
+
+execute_process(COMMAND uname -m OUTPUT_VARIABLE MACHINE_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if(LSB_RELEASE)
+    execute_process(COMMAND lsb_release -sr OUTPUT_VARIABLE LINUX_DISTRO_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+    execute_process(COMMAND lsb_release -si OUTPUT_VARIABLE LINUX_DISTRO OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+    string(TOLOWER ${LINUX_DISTRO} LINUX_DISTRO)
+endif()
+
+# default CPack generators
+set(CPACK_GENERATOR TGZ STGZ)
+
+# extra CPack generators
+if(LINUX_DISTRO STREQUAL "debian" OR LINUX_DISTRO STREQUAL "ubuntu" OR LINUX_DISTRO STREQUAL "linuxmint")
+    list(APPEND CPACK_GENERATOR DEB)
+elseif(LINUX_DISTRO STREQUAL "fedora" OR LINUX_DISTRO STREQUAL "opensuse" OR LINUX_DISTRO STREQUAL "centos")
+    list(APPEND CPACK_GENERATOR RPM)
+endif()
+
+# general CPack config
+set(CPACK_PACKAGE_DIRECTORY ${CMAKE_BINARY_DIR}/packages)
+message(STATUS "CPack: packages will be generated under ${CPACK_PACKAGE_DIRECTORY}")
+set(CPACK_PACKAGE_NAME "lib${PROJECT_NAME}")
+set(CPACK_PACKAGE_VERSION ${PROJECT_VERSION})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "LibTomMath")
+set(CPACK_PACKAGE_VENDOR "LibTomMath")
+set(CPACK_PACKAGE_CONTACT "libtom@googlegroups.com")
+set(CPACK_RESOURCE_FILE_LICENSE "${PROJECT_SOURCE_DIR}/LICENSE")
+set(CPACK_PACKAGE_FILE_NAME ${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${PACKAGE_RELEASE_VERSION}_${LINUX_DISTRO}-${LINUX_DISTRO_VERSION}_${MACHINE_ARCH})
+set(CPACK_STRIP_FILES ON)
+
+# deb specific CPack config
+set(CPACK_DEBIAN_PACKAGE_RELEASE ${PACKAGE_RELEASE_VERSION})
+set(CPACK_DEBIAN_FILE_NAME DEB-DEFAULT)
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "https://www.libtom.net/LibTomMath")
+set(CPACK_DEBIAN_PACKAGE_SECTION "devel")
+
+# rpm specific CPack config
+set(CPACK_RPM_PACKAGE_URL "https://www.libtom.net/LibTomMath")
+set(CPACK_RPM_PACKAGE_RELEASE ${PACKAGE_RELEASE_VERSION})
+set(CPACK_RPM_PACKAGE_ARCHITECTURE ${MACHINE_ARCH})
+set(CPACK_RPM_PACKAGE_NAME "lib${PROJECT_NAME}-${PROJECT_VERSION}")
+set(CPACK_RPM_FILE_NAME "lib${PROJECT_NAME}_${PROJECT_VERSION}-${CPACK_RPM_PACKAGE_RELEASE}_${LINUX_DISTRO}-${LINUX_DISTRO_VERSION}_${CPACK_RPM_PACKAGE_ARCHITECTURE}.rpm")
+set(CPACK_RPM_PACKAGE_LICENSE "WTFPL")
+
+include(CPack)
```

Comment:
There is also a [CPack BSD](https://cmake.org/cmake/help/latest/cpack_gen/freebsd.html) for the `pkg` format but it has been added in CMake 3.10

Reply:
I guess we could up our minimum CMake requirements as 3.7 is pretty old. If we take "everyone's preferred Distro"^TM as reference, Debian Stretch is the only still-maintained version that brings only 3.7, all others bring at least 3.10. Stretch-backports also brings 3.16, so we should be fine with upping to 3.10.

Reply:
> I guess we could up our minimum CMake requirements as 3.7 is pretty old. If we take "everyone's preferred Distro"^TM as reference, Debian Stretch is the only still-maintained version that brings only 3.7, all others bring at least 3.10. Stretch-backports also brings 3.16, so we should be fine with upping to 3.10.

@czurnieden @mrpelotazo your thoughts on this?

Reply:
Looking at the BSD's: FreeBSD has 3.5 for their oldest  version 11 but only for [selected architectures](https://www.freshports.org/devel/cmake), [OpenBSD has 3.19](https://ftp.openbsd.org/pub/OpenBSD/6.9/packages/) in its oldest version 6.9, NetBSD seems to have the newest with [3.22](http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/devel/cmake/). I didn't check the more obscure OSs LTM supports but I think were on the safe side with CMake 3.10

Reply:
Cpack FreeBSD works in FreeBSD 13 (CPack needs to be included manually when building cmake from ports but every FreeBSD developer should know that. I did not ;-) ) although we forgot to add it to the list. See [my fork of this branch](https://github.com/czurnieden/libtommath/tree/add-cmake-support)

Reply:
@sjaeckel well I don't know how far back you guys want to go. Version 3.10 sounds reasonable to me.

Reply:
FYI I've added the FreeBSD section to CPack even though it doesn't work... CPack support in FreeBSD's CMake is marked as experimental and after enabling it and re-building cmake, it segfaults if executed ...

```
$ cpack -G "FREEBSD"
CPack: Create package using FREEBSD
CPack: Install projects
CPack: - Run preinstall target for: tommath
CPack: - Install project: tommath []
CPack: Create package
Segmentation fault (core dumped)
```

CC @gahr @lbartoletti

Reply:
Sorry I have no experience with CPath and I haven't followed this discussion. What do we need it for?

Reply:
Oh, I just read it up. Can we turn it off on FreeBSD? We don't use it to generate packages anyway.